src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigStorageLocation.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeStyle;
+
+namespace Microsoft.CodeAnalysis.Options
+{
+    internal static class EditorConfigStorageLocation
+    {
+        public static EditorConfigStorageLocation<bool> ForBoolOption(string keyName)
+            => new EditorConfigStorageLocation<bool>(keyName, s_parseBool);
+
+        public static EditorConfigStorageLocation<int> ForInt32Option(string keyName)
+            => new EditorConfigStorageLocation<int>(keyName, s_parseInt32);
+
+        public static EditorConfigStorageLocation<CodeStyleOption<bool>> ForBoolCodeStyleOption(string keyName)
+            => new EditorConfigStorageLocation<CodeStyleOption<bool>>(keyName, s_parseBoolCodeStyleOption);
+
+        private static Func<string, Optional<bool>> s_parseBool = ParseBool;
+        private static Optional<bool> ParseBool(string str)
+            => bool.TryParse(str, out var result) ? result : new Optional<bool>();
```

Comment:
would like to just have the field initialized with a lambda.  But out-vars are not supported there due to a compiler bug we shipped in Dev15.  When we get an updated compiler this can be simplified.